File: 其他/reverseWords/reverseWords.java
class Solution {
    public String reverseWords(String s) {
        int left=0,right=s.length()-1;
        while(s.charAt(left) == ' ') left++;
        while(s.charAt(right) == ' ') right--;
        Deque<String> dq = new ArrayDeque<String>();
        StringBuilder tmp = new StringBuilder();
        for(int i=left;i<=right;i++){
            char ch = s.charAt(i);
            if(ch == ' '){
                if(tmp.length() != 0) dq.offerFirst(tmp.toString());
                tmp.setLength(0);
            }
            else{
                tmp.append(ch);
            }
        }
        if(tmp.length() != 0) dq.offerFirst(tmp.toString());
        return String.join(" ", dq);
    }
}

File: 其他/reverseWords/reverseWords.py
# 反转字符串，将两头的空格去掉
# python有个很粗暴的解法，一行即可以解决
# 其他解法如提取出字符串，存储在栈中，然后pop，在java和c中实现


class Solution:
    def reverseWords(self, s: str) -> str:
        return " ".join(reversed(s.split()))